srcDLL/DllMain.cpp

```diff
@@ -98,13 +98,15 @@ void OnInit()
 
 	// Find VOL files from additional folders
 	for (std::size_t i = 0; i < moduleLoader->Count(); ++i) {
-		std::string moduleDirectory = moduleLoader->GetModuleDirectory(i);
+		const auto moduleDirectory = moduleLoader->GetModuleDirectory(i);
```

Comment:
Maybe best to use `const auto&`.

Reply:
In retrospect, this is probably best not being a reference. There is no guarantee a module doesn't just generate the directory and return it without storing a local copy.

Reply:
C++ has lifetime extension rules to handle cases like that. It's not a problem. A returned temporary object will not go out of scope and be destructed until after the capturing reference goes out of scope.

Reply:
OK, I didn't realize that was the case. I think the tip of the weeks would not recommend implementing this. Although, now that I know it is legal, I guess I don't care too much one way or the other. 

From https://abseil.io/tips/101

> Scenario #8 is what allows most reflexive use of references (that is “Oh, I don’t want to copy so I’ll just assign into a reference” without necessarily thinking about what is being returned). However, because of #1, it’s also not really doing anything for you: there probably wouldn’t have been a copy in the first place. Further, now readers of your code have to contend with your local variable being of type const string& instead of string, and thus worry about whether the underlying string has gone out of scope or changed.